src/diffusers/image_processor.py

```diff
@@ -165,17 +172,39 @@ def preprocess(
 
     def postprocess(
         self,
-        image,
+        image: torch.FloatTensor,
         output_type: str = "pil",
+        do_denormalize: Optional[List[bool]] = None,
     ):
-        if isinstance(image, torch.Tensor) and output_type == "pt":
+        if not isinstance(image, torch.Tensor):
+            raise ValueError(
+                f"Input for postprocess is in incorrect format: {type(image)}.  we only support pytorch tensor"
+            )
+        if output_type not in ["latent", "pt", "np", "pil"]:
+            deprecation_message = (
+                f"the output_type {output_type} is outdated and has been set to `np`. Please make sure to set it to one of these instead: "
+                "`pil`, `np`, `pt`, `latent`"
+            )
+            deprecate("Unsupported output_type", "1.0.0", deprecation_message, standard_warn=False)
+            output_type = "np"
+
+        if output_type == "latent":
+            return image
+
+        if do_denormalize is None:
+            do_denormalize = [self.config.do_normalize] * image.shape[0]
+
+        image = torch.stack(
+            [self.denormalize(image[i]) if do_denormalize[i] else image[i] for i in range(image.shape[0])]
```

Comment:
This is neat!

But would there be a need to denormalize images individually? I would assume if `do_denormalize` is True then ALL the images will be denormalized. Why individually? 

Nit: Should it be "unnormalize"?

Reply:
normally we should denormalize all images if we have normalized them in preprocessing (i.e. `if self.config.do_normalize is True`)
however, there is this edge case when the image is black due to `safety_checker` - in that case, we will pass a `do_denormalize` argument to allow skip the denormalize process only for these nsfw images.

Reply:
Superb! Thanks for explaining!